$OpenBSD: patch-lib_zip_dirent_c,v 1.1 2015/04/03 18:20:49 sthen Exp $

Int overflow CVE-2015-2331 - just using reallocarray here; upstream's commit
http://hg.nih.at/libzip/rev/9f11d54f692e builds on other changes.

--- lib/zip_dirent.c.orig	Thu Nov 28 16:57:10 2013
+++ lib/zip_dirent.c	Fri Apr  3 19:16:42 2015
@@ -110,7 +110,7 @@ _zip_cdir_new(zip_uint64_t nentry, struct zip_error *e
 
     if (nentry == 0)
 	cd->entry = NULL;
-    else if ((cd->entry=(struct zip_entry *)malloc(sizeof(*(cd->entry))*(size_t)nentry)) == NULL) {
+    else if ((cd->entry=(struct zip_entry *)reallocarray(NULL, sizeof(*(cd->entry)),(size_t)nentry)) == NULL) {
 	_zip_error_set(error, ZIP_ER_MEMORY, 0);
 	free(cd);
 	return NULL;
